src_python/habitat_sim/simulator.py

```diff
@@ -576,6 +587,10 @@ def __init__(self, sim: Simulator, agent: Agent, sensor_id: str) -> None:
 
         self._spec = self._sensor_object.specification()
 
+        # When using the batch renderer, no memory is allocated here.
+        if self._sim.config.enable_batch_renderer:
```

Comment:
Feels a bit dangerous to just return out of the function early. How about wrap all the renderer code below in a helper function and only call that for `not enable_batch_renderer`.

Also, is it safe/correct to skip the noise model piece below for the batch renderer case?

Reply:
> How about wrap all the renderer code below in a helper function and only call that for not enable_batch_renderer

Sounds good!

> Also, is it safe/correct to skip the noise model piece below for the batch renderer case?

The noise modeling has to be done by the batch renderer. I think that it could make sense to preserve the noise model function loading here and call it from Habitat-Lab after rendering.